pkg/common/utils.go

```diff
@@ -17,3 +23,44 @@ func RemoveStringSliceItem(item string, slice *[]string) {
 		}
 	}
 }
+
+// ParseResponseForKeywords parses the response from detector verification calls for expected keywords in the response.
+func ParseResponseForKeywords(reader io.ReadCloser, keywords []string) (bool, error) {
+	for _, keyword := range keywords {
+		if keyword == "" {
+			continue
+		}
+
+		found, err := containsSubstring(reader, keyword)
+
+		if err != nil {
+			return false, err
+		}
+		return found, nil
+	}
+
+	return false, nil
+}
+
+func containsSubstring(reader io.ReadCloser, target string) (bool, error) {
```

Comment:
suggestion: You can probably even use the `bufio.NewScanner` here. It's a little easier to read imo.
```
func containsSubstring(reader io.Reader, substring string) (bool, error) {
	scanner := bufio.NewScanner(reader)
	for scanner.Scan() {
		if strings.Contains(scanner.Text(), substring) {
			return true, nil
		}
	}
	if err := scanner.Err(); err != nil {
		return false, err
	}
	return false, nil
}
```

Reply:
thanks for the suggestion!